tools/testing/source_code_test.py

```diff
@@ -147,6 +147,7 @@ def test_no_experimental_api():
     # This allowlist should not grow. Do not add elements to this list.
     allowlist = [
         "tensorflow_addons/optimizers/weight_decay_optimizers.py",
+        "tensorflow_addons/layers/max_unpooling_2d.py",
```

Comment:
Gentle ping to @seanpmorgan. I think we should relax the constraints to experimental APIs as we only pin to stable tensorflow version. More pros than cons I believe.

Reply:
Can we confirm that experimental APIs are aliased to their final API once they move out of experimental? If so then I think its okay, but I'd still like to track the usage in this source code test since it should be discouraged when possible. The API parameters are still able to change when graduating

Reply:
cc @thaink for API questions. If so, I think we should also add test files into source code tests. Like mixed precision API in our tests, they do change some parameters/underlying implementation from experimental to stable one, but we just do not use them so there is no regression issue.

Reply:
@WindQAQ Can you point me to that test file?
The experimental_implements here is only used for tflite conversion so it is not possible to add tests about that at this point since the conversion code is dependent on this. Besides, since it will only affect tflite users if changed, I think it might be better to add the tests on our side.

Reply:
Sorry for the confusion. It's unrelated to this PR. We have used some experimental API here. `run_functions_eagerly` since 2.2 and `mixed_precision` since 2.3, but because they are used in unit tests, we're not hostile to them.

https://github.com/tensorflow/addons/blob/master/tensorflow_addons/utils/test_utils.py

Reply:
Thanks @WindQAQ.
Is there anything more I should do before submitting this PR?

Reply:
No, let's wait for @seanpmorgan 's response. Thanks again for the patience.